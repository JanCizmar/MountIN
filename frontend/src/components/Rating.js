import React from 'react';
import StarRatingComponent from 'react-star-rating-component';

export default function Rating(props) {
    return <StarRatingComponent
        {...props}
        name="rating"
        starCount={5}
        starColor="#158a00"
    />
}