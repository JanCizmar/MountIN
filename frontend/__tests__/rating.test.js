import React from 'react';
import renderer from 'react-test-renderer';
import Rating from "../src/components/Rating";

it('renders correctly', () => {
    const tree = renderer
        .create(<Rating value={2}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
