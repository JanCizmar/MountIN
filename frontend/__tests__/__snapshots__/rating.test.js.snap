// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="dv-star-rating"
  style={
    Object {
      "display": "inline-block",
      "position": "relative",
    }
  }
>
  <input
    checked={false}
    className="dv-star-rating-input"
    id="rating_5"
    name="rating"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
        "marginLeft": -9999,
        "position": "absolute",
      }
    }
    type="radio"
    value={5}
  />
  <label
    className="dv-star-rating-star dv-star-rating-empty-star"
    htmlFor="rating_5"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "color": "#333",
        "cursor": "pointer",
        "float": "right",
      }
    }
  >
    <i
      style={
        Object {
          "fontStyle": "normal",
        }
      }
    >
      ★
    </i>
  </label>
  <input
    checked={false}
    className="dv-star-rating-input"
    id="rating_4"
    name="rating"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
        "marginLeft": -9999,
        "position": "absolute",
      }
    }
    type="radio"
    value={4}
  />
  <label
    className="dv-star-rating-star dv-star-rating-empty-star"
    htmlFor="rating_4"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "color": "#333",
        "cursor": "pointer",
        "float": "right",
      }
    }
  >
    <i
      style={
        Object {
          "fontStyle": "normal",
        }
      }
    >
      ★
    </i>
  </label>
  <input
    checked={false}
    className="dv-star-rating-input"
    id="rating_3"
    name="rating"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
        "marginLeft": -9999,
        "position": "absolute",
      }
    }
    type="radio"
    value={3}
  />
  <label
    className="dv-star-rating-star dv-star-rating-empty-star"
    htmlFor="rating_3"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "color": "#333",
        "cursor": "pointer",
        "float": "right",
      }
    }
  >
    <i
      style={
        Object {
          "fontStyle": "normal",
        }
      }
    >
      ★
    </i>
  </label>
  <input
    checked={true}
    className="dv-star-rating-input"
    id="rating_2"
    name="rating"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
        "marginLeft": -9999,
        "position": "absolute",
      }
    }
    type="radio"
    value={2}
  />
  <label
    className="dv-star-rating-star dv-star-rating-full-star"
    htmlFor="rating_2"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "color": "#158a00",
        "cursor": "pointer",
        "float": "right",
      }
    }
  >
    <i
      style={
        Object {
          "fontStyle": "normal",
        }
      }
    >
      ★
    </i>
  </label>
  <input
    checked={false}
    className="dv-star-rating-input"
    id="rating_1"
    name="rating"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
        "marginLeft": -9999,
        "position": "absolute",
      }
    }
    type="radio"
    value={1}
  />
  <label
    className="dv-star-rating-star dv-star-rating-full-star"
    htmlFor="rating_1"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "color": "#158a00",
        "cursor": "pointer",
        "float": "right",
      }
    }
  >
    <i
      style={
        Object {
          "fontStyle": "normal",
        }
      }
    >
      ★
    </i>
  </label>
</div>
`;
